src/react-components/presence-log.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styles from "../assets/stylesheets/presence-log.scss";
+import classNames from "classnames";
+import Linkify from "react-linkify";
+import { emojify } from "react-emojione";
+import { FormattedMessage } from "react-intl";
+
+export default class PresenceLog extends Component {
+  static propTypes = {
+    entries: PropTypes.array,
+    inRoom: PropTypes.bool
+  };
+
+  constructor(props) {
+    super(props);
+  }
+
+  domForEntry = e => {
+    const entryClasses = {
+      [styles.presenceLogEntry]: true,
+      [styles.expired]: !!e.expired
+    };
+
+    switch (e.type) {
+      case "join":
+      case "entered":
+        return (
+          <div key={e.key} className={classNames(entryClasses)}>
+            <b>{e.name}</b> <FormattedMessage id={`presence.${e.type}_${e.presence}`} />
+          </div>
+        );
+      case "leave":
+        return (
+          <div key={e.key} className={classNames(entryClasses)}>
+            <b>{e.name}</b> <FormattedMessage id={`presence.${e.type}`} />
+          </div>
+        );
+      case "display_name_changed":
+        return (
+          <div key={e.key} className={classNames(entryClasses)}>
+            <b>{e.oldName}</b> <FormattedMessage id="presence.name_change" /> <b>{e.newName}</b>.
+          </div>
+        );
+      case "message":
+        return (
+          <div key={e.key} className={classNames(entryClasses)}>
+            <b>{e.name}</b>:{" "}
+            <Linkify properties={{ target: "_blank", rel: "noopener referrer" }}>
```

Comment:
Linkify parses emails and mailto: links. Should be fine, unless we want to prevent that explicitly